jupyterhub_carina/CarinaOAuthClient.py

```diff
@@ -98,27 +99,70 @@ def create_cluster(self, cluster_name):
         """
         Create a Carina cluster
         """
-        self.log.info("Creating cluster %s/%s", self.user, cluster_name)
+        template_id = yield self.lookup_swarm_template()
+
+        self.log.info("Creating cluster %s/%s using template %d", self.user, cluster_name,
+                      template_id)
         request = HTTPRequest(
-            url=os.path.join(self.CARINA_CLUSTERS_URL, cluster_name),
-            method='PUT',
-            body='{}',
+            url=self.CARINA_CLUSTERS_URL,
+            method='POST',
+            body=json.dumps({
+                      "cluster_type_id": template_id,
+                      "node_count": 1,
+                      "name": cluster_name
+                      }),
             headers={
+                'Content-Type': 'application/json',
                 'Accept': 'application/json'
             })
+        self.log.info("Request: %s", request.body)
 
         response = yield self.execute_oauth_request(request)
         result = json.loads(response.body.decode('utf8', 'replace'))
+        self.log.info("Response: %s", response.body)
+
         return result
 
     @gen.coroutine
-    def download_cluster_credentials(self, cluster_name, destination, polling_interval=30):
+    def lookup_swarm_template(self):
```

Comment:
Templates define what the cluster should look like which at the moment is just picking "Kubernetes 1.5" or "Swarm 1.11.2".  It doesn't tie back to heat, and acts as a friendly name for an ansible playbook commit hash on the backend.

For magnum, I _think_ they do use heat templates.